src/Formats/JSONUtils.cpp

```diff
@@ -523,6 +550,45 @@ namespace JSONUtils
         }
     }
 
+    void writeCompactAdditionalInfo(
+        size_t rows,
+        size_t rows_before_limit,
+        bool applied_limit,
+        const Stopwatch & watch,
+        const Progress & progress,
+        bool write_statistics,
+        WriteBuffer & out)
+    {
+        writeCompactObjectStart(out);
+        writeCompactObjectStart(out, 0, "statistics");
+        writeTitle("rows", out, 0, "");
+        writeIntText(rows, out);
+        writeFieldCompactDelimiter(out);
```

Comment:
Trailing comma if nothing is written (in case of exception):

```
$ echo "Select throwIf(number = 15), 1::Int64 as a, '\"' from numbers(100) format JSONCompactWithProgress settings output_format_json_quote_64bit_integers=1, max_block_size=10" | curl 'localhost:8123/?max_threads=1' --data-binary @-
{"meta": [{"name":"throwIf(equals(number, 15))", "type":"UInt8"}, {"name":"a", "type":"Int64"}, {"name":"'\"'", "type":"String"}]}}
{"progress":{"read_rows":"0","read_bytes":"0","written_rows":"0","written_bytes":"0","total_rows_to_read":"0","result_rows":"0","result_bytes":"0","elapsed_ns":"0"}}
{"progress":{"read_rows":"0","read_bytes":"0","written_rows":"0","written_bytes":"0","total_rows_to_read":"100","result_rows":"0","result_bytes":"0","elapsed_ns":"0"}}
{"progress":{"read_rows":"10","read_bytes":"80","written_rows":"0","written_bytes":"0","total_rows_to_read":"100","result_rows":"0","result_bytes":"0","elapsed_ns":"0"}}
{"data":[0, "1", "\""]}
{"data":[0, "1", "\""]}
{"data":[0, "1", "\""]}
{"data":[0, "1", "\""]}
{"data":[0, "1", "\""]}
{"data":[0, "1", "\""]}
{"data":[0, "1", "\""]}
{"data":[0, "1", "\""]}
{"data":[0, "1", "\""]}
{"data":[0, "1", "\""]}
{"progress":{"read_rows":"20","read_bytes":"160","written_rows":"0","written_bytes":"0","total_rows_to_read":"100","result_rows":"0","result_bytes":"0","elapsed_ns":"0"}}
{"statistics": {"rows":10, }}
{"exception": "Code: 395. DB::Exception: Value passed to 'throwIf' function is non-zero: while executing 'FUNCTION throwIf(equals(__table1.number, 15_UInt8) :: 0) -> throwIf(equals(__table1.number, 15_UInt8)) UInt8 : 1'. (FUNCTION_THROW_IF_VALUE_IS_NON_ZERO) (version 24.8.1.232)"}
```

In this case `"statistics"` should not be written